apps/functions-subscription/src/domain/users.ts

```diff
@@ -0,0 +1,13 @@
+interface BaseUser {
+  readonly id: string;
+}
+
+type TrialOwner = BaseUser & {
```

Comment:
**note**: here we are defining the type of user of the trial system.

at the moment there are the trial manager (or owner) and the user who can subscribe to a trial (the subscriber)

Reply:
What's the purpose of the `type` field? Can't we use user groups directly?

Reply:
`type` field will be used to discriminate between users.
IMHO this helps to decouple something coming from the adapters layer (like the user groups) to something that the system should handle and it's the core of the system.